smdebug/tensorflow/keras.py

```diff
@@ -550,12 +550,28 @@ def on_test_batch_begin(self, batch, logs=None):
     def on_predict_batch_begin(self, batch, logs=None):
         self._on_any_batch_begin(batch, ModeKeys.PREDICT, logs=logs)
 
+    def _write_optimizer_variables(self):
+        optimizer_collections = self.collection_manager.get(CollectionKeys.OPTIMIZER_VARIABLES)
+        if optimizer_collections in self._get_collections_to_save_for_step():
+            for tensor_ref in optimizer_collections.get_tensors(self.mode):
+                tensor = tensor_ref.tf_obj
+                self._save_for_tensor(
+                    tensor_name=tensor.name, tensor_value=tensor.value(), check_before_write=False
+                )
+
     def _on_any_batch_end(self, batch, mode, logs=None):
         if self._is_not_supported():
             return
 
         if not is_tf_version_2x() or (is_tf_version_2x() and not tf.executing_eagerly()):
             self._remove_fetches_and_callbacks(mode)
+
+        if is_tf_version_2x() and tf.executing_eagerly():
+            # Need to prepare non layer tensors again since
+            # some tensors only become available on  batch end
+            self._prepare_non_layer_tensors()
+            self._write_optimizer_variables()
```

Comment:
if these calls are made here, they must be executed for vanilla as well as custom TF. right?

custom TF also makes a call to set_optimizer_variables() ? does it try to set the same set of tensors or different? do we need an additional set_optimizer_variables() call if we get all the variables here?

Reply:
There is no difference in tensors saved when run on AWS TF v/s vanilla